### PR TITLE
PIM-9395: Merge 4.0 into master

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by SamirBoulil — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Component/Product/Factory/EmptyValuesCleaner.php [hunk 1]
```diff
@@ -56,6 +56,13 @@ private function isFilled($data): bool
         }
 
         if (is_array($data)) {
+            // @see PIM-9112: Fix publish product if metric with null values
+            // @todo to rework in 5.0:
```
Comment (rouliane) [inline src/Akeneo/Pim/Enrichment/Component/Product/Factory/EmptyValuesCleaner.php:60]: todo to handle.

**Review by rouliane — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Component/Product/Factory/EmptyValuesCleaner.php (hunk 1 above)
Comment (rouliane) [reply src/Akeneo/Pim/Enrichment/Component/Product/Factory/EmptyValuesCleaner.php:60]: or not. I'm not sure.

Comment (SamirBoulil) [reply src/Akeneo/Pim/Enrichment/Component/Product/Factory/EmptyValuesCleaner.php:60]: huh, let me know because i'm starting to fix it xD

**Comment by anaisbaunelemaire** [comment]
check unfreeze

**Comment by anaisbaunelemaire** [comment]
check unfreeze

**Pull request merged**